lib/parallel_tests/grouper.rb

```diff
@@ -28,7 +28,39 @@ def in_even_groups_by_size(items, num_groups, options = {})
           raise 'Number of isolated processes must be less than total the number of processes'
         end
 
-        if isolate_count >= 1
+        if options[:specify_groups]
+          specify_test_process_groups = options[:specify_groups].split('|')
+          all_specified_tests = specify_test_process_groups.map { |group| group.split(',') }.flatten
+          if specify_test_process_groups.count > num_groups
+            raise 'Number of processes separated by pipe must be less than or equal to the total number of processes'
+          end
+
+          specified_items_found, items = items.partition do |item, _size|
+            all_specified_tests.any? { |specified_spec| item == specified_spec }
+          end
```

Comment:
```suggestion
          specified_items_found = items.select! { |item, _size| all_specified_tests.include?(item) } || []
```

Reply:
Don't we want them partitioned? I support using partition because, we will have extracted the specified tests from the items array, then we can order the specified tests as they are specified, and, order the items in the remaining parallel processes.

By using the `!` we'd be losing the other tests that aren't specified in specified_groups, the ones that will be ordered in the remaining parallel processes